### ISSUE TITLE
TSLint expect trailing comma on multiline function args, which is invalid JavaScript

### ISSUE THREAD

**Issue by rcchen** [palantir/tslint#2367]
### Bug Report

- __TSLint version__: 4.5.1
- __TypeScript version__: 2.1.6
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
const arr = [];
arr.push(
  "foo",
  "bar"
);
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint:latest", "tslint-react"]
}
```

#### Actual behavior

TSLint expects trailing comma after the last argument to `arr.push`

#### Expected behavior

TSLint should not expect trailing comma after the last argument as that's not valid JS syntax.

**Comment by ajafff** [comment]
> as that's not valid JS syntax

But it will eventually be valid JS syntax. http://2ality.com/2015/11/trailing-comma-parameters.html

It's also valid typescript syntax. And that's what this package is made for.
If you don't like the default config provided by tslint, you can still disable the rule or set it to `"never"` to ban trailing commas completely. https://palantir.github.io/tslint/rules/trailing-comma/

**Comment by rcchen** [comment]
Sure, but `tslint` picked up the ability to lint JS code in `4.0.0`, and as of date, it's not valid JS syntax.

Given a mixed JS/TS codebase where the linter is being applied across both types of files, there's an issue present.

**Comment by adidahiya** [comment]
@rcchen you're right; the best thing you can do for now is have a separate `tslint.json` file for linting JS that extends your main tslint config and disables the `trailing-comma` rule. I see how that's non-ideal since you miss out on object & array literal checking. I think we should eventually make rules smarter about these kinds of things in the rare cases where JS / TS syntax diverges

**Comment by adidahiya** [comment]
I don't think #2367 fixes this exactly. If someone _does_ want to lint for trailing commas in multiline function args in TS but also use the same config for JS files, they won't be able to do so with the built-in configs.

**Comment by adidahiya** [comment]
Actually, I like @ajafff's suggestion for the solution (https://github.com/palantir/tslint/issues/2775#issuecomment-302139330). Just use the `jsRules` block to override lint behavior in JS files: 

```json
{
    "jsRules": {
        "trailing-comma": [true, {"multiline": {"arguments": "never", ...}}]
    }
}
```
